src/components/AddContactForm.js

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { REGEX } from '../constants/constants';
+import { validateInput } from '../helpers/helper';
+import './AddContactForm.css';
+
+class AddContactForm extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstName: '',
+      lastName: '',
+      imageUrl: '',
+      firstNameError: false,
+      lastNameError: false,
+      imageUrlError: false
+    };
+  }
+
+  onFirstNameInputChange = (event) => {
+    const { target } = event;
+    const newFirstName = target.value.trim();
+    this.setState({ firstName: newFirstName }, () => {
+      const { firstName } = this.state;
+      this.setState({ firstNameError: !validateInput(firstName, REGEX.NAME) });
```

Comment:
i don't think having a setState inside a setState is the best idea. can't you do this with a single setState?
(same for the next 2 methods)